waspc/data/Generator/templates/react-app/src/operations/resources.js

```diff
@@ -45,3 +45,23 @@ export const invalidateQueriesUsing = (resources) => {
     queryClient.invalidateQueries(queryCacheKey)
   }
 }
+
+export function removePrivateQueries() {
+  // TODO(matija): Currently we are removing all the queries, but we should remove only
+  // non-public, user-dependent queries - public queries are expected not to change in respect
+  // to the currently logged in user.
+  // TODO(filip): why remove? What does it do exactly: https://github.com/tannerlinsley/react-query/blob/master/src/core/queryClient.ts
+  queryClient.removeQueries()
+}
+
+export function invalidateAndClearQueries() {
+  // TODO(matija): We are currently invalidating all the queries, but we should clear only the
+  // non-public, user-dependent ones.
+  // TODO(filip): I don't think we should be invalidating anything here
+  queryClient.invalidateQueries()
+
+  // TODO(matija): We are currently clearing all the queries, but we should clear only the
+  // non-public, user-dependent ones.
+  // TODO(filip): why clear? What does it do exactly: https://github.com/tannerlinsley/react-query/blob/master/src/core/queryClient.ts
```

Comment:
I'll make sure to get to the bottom of this before merging, no worries :)

Reply:
I documented the reasoning behind all operations.

By the way, it turns out that `queryClient.removeQueries()` does the same thing as `queryClient.clear()`. I changed both usages to `queryClient.removeQueries()` because `removeQueries` is more flexible (i.e., it takes a filter).